parlai/tasks/wrapper/agents.py

```diff
@@ -144,3 +105,58 @@ def shutdown(self):
 
     def update_counters(self):
         self.task.update_counters()
+
+
+class LabelToTextTeacher(AbstractWrapperTeacher):
+    """
+    Teacher that will shift message['labels'][0] into message['text'] for whatever task
+    is specified with --label-to-text-task.
+
+    Because the dialogue history is effectively overwritten by this action, all episodes
+    will be flattened into one example each.
+    """
+
+    @classmethod
+    def add_cmdline_args(cls, argparser):
+        # Teacher has no args
+        parser = argparser.add_argument_group('LabelToText args')
+        parser.add_argument(
+            '-l2tt',
+            '--label-to-text-task',
+            type=str,
+            help='The task whose labels will get shifted into the text field',
+        )
+
+    def __init__(self, opt: Opt, shared=None):
+        if ',' in opt['task']:
+            raise ValueError('LabelToTextTeacher cannot be used with multiple tasks!')
+        self.id = opt['task']
+        self.opt = opt
+        if shared and 'task' in shared:
+            self.task = create_agent_from_shared(shared['task'])
+        else:
+            opt_singletask = copy.deepcopy(opt)
+            opt_singletask['task'] = opt['label_to_text_task']
+            self.task = create_task_agent_from_taskname(opt_singletask)[0]
+
+
+    def act(self):
+        """
+        Act on the previous observation.
+        """
+        act = self.task.act()
+        new_act = copy.deepcopy(act)
+        if 'labels' in act:
```

Comment:
can condense these two blocks since only 'labels' or 'eval_labels' will ever be in act at a given time. `labels = 'labels' if 'labels' in act else 'eval_labels'`